imports/server/onExit.ts

```diff
@@ -0,0 +1,12 @@
+const exitHandlers: (() => void)[] = [];
+
+['SIGINT' as const, 'SIGTERM' as const, 'SIGHUP' as const].forEach((signal) => {
+  process.once(signal, () => {
```

Comment:
Checking to be sure I'm understanding this logic:

For each signal type, we register an event handler to be run exactly once upon delivery of that signal.

Upon the first such signal to arrive, the event handler takes the list of exit handlers, and runs each cleanup function in the order they were registered, exactly once.

And then, the event handler signals itself *again* with the original signal, which, because we've no longer got a handler for those signals registered, ultimately causes the process to exit.

Is that right?

Reply:
Yep, that's right